server/shared/src/main/scala/org/http4s/server/middleware/Throttle.scala

```diff
@@ -55,37 +58,47 @@ object Throttle {
 
     /** Creates an in-memory [[TokenBucket]].
       *
-      * @param capacity the number of tokens the bucket can hold and starts with.
+      * @param capacity    the number of tokens the bucket can hold and starts with.
       * @param refillEvery the frequency with which to add another token if there is capacity spare.
       * @return A task to create the [[TokenBucket]].
       */
     def local[F[_]](capacity: Int, refillEvery: FiniteDuration)(implicit
         F: Temporal[F]
     ): F[TokenBucket[F]] = {
-      def getTime = F.monotonic.map(_.toNanos)
-      val bucket = getTime.flatMap(time => F.ref((capacity.toDouble, time)))
-
-      bucket.map { counter =>
-        new TokenBucket[F] {
-          override def takeToken: F[TokenAvailability] =
-            counter.access.flatMap { case ((previousTokens, previousTime), setter) =>
-              getTime.flatMap { currentTime =>
-                val timeDifference = currentTime - previousTime
-                val tokensToAdd = timeDifference.toDouble / refillEvery.toNanos.toDouble
-                val newTokenTotal = Math.min(previousTokens + tokensToAdd, capacity.toDouble)
-                def unavailable = {
-                  val timeToNextToken = refillEvery.toNanos - timeDifference
+      val getNanoTime = Temporal[F].monotonic.map(_.toNanos)
+      for {
+        nanoValue <- refillEvery.toNanos.pure[F]
+        // make sure that refillEvery is positive
+        _ <- F.raiseUnless(nanoValue > 0L)(
```

Comment:
I promise one last nit 😬 

```suggestion
      val getNanoTime = Temporal[F].monotonic.map(_.toNanos)
      val refillEveryNanos = refillEvery.toNanos
      for {
        // make sure that refillEvery is positive
        _ <- F.raiseUnless(nanoValue > 0L)(
```